rust_icu_uloc/src/lib.rs

```diff
@@ -101,6 +112,80 @@ impl ULoc {
         ffi::CString::new(self.repr.clone()).expect("ULoc contained interior NUL bytes")
     }
 
+    pub fn accept_language(
+        accept_list: impl IntoIterator<Item = impl Into<ULoc>>,
+        available_locales: impl IntoIterator<Item = impl Into<ULoc>>,
+    ) -> Result<(Option<ULoc>, UAcceptResult), common::Error> {
+        let mut buf: Vec<u8> = vec![0; LOCALE_CAPACITY];
+        let mut accept_result: UAcceptResult = UAcceptResult::ULOC_ACCEPT_FAILED;
+        let mut status = common::Error::OK_CODE;
+
+        let mut accept_list_cstrings: Vec<ffi::CString> = vec![];
+        // This is mutable only to satisfy the missing `const`s in the ICU4C API.
+        let mut accept_list: Vec<*const raw::c_char> = accept_list
+            .into_iter()
+            .map(|item| {
+                let uloc: ULoc = item.into();
+                accept_list_cstrings.push(uloc.as_c_str());
+                accept_list_cstrings.last().expect("non-empty list").as_ptr()
```

Comment:
I wonder if we should remove the method as_c_str from `ULoc`, and introduce an as_ref instead, to avoid the issue with not recording the temporary?

That should make it impossible to step onto the issue due to lifetime requirements, and whoever needs an independent copy should clone `ULoc`.

Reply:
Since we're undecided on what this API should look like, let's leave the change for a separate CL.